src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -46,8 +46,13 @@ public LogicManager(Model model, Storage storage) {
     public CommandResult execute(String commandText) throws CommandException, ParseException {
         logger.info("----------------[USER COMMAND][" + commandText + "]");
 
-        CommandResult commandResult;
         Command command = addressBookParser.parseCommand(commandText);
+        return execute(command);
+    }
+
+    @Override
+    public CommandResult execute(Command command) throws CommandException {
+        CommandResult commandResult;
         commandResult = command.execute(model);
```

Comment:
Can this be simplified to CommandResult commandResult = command.execute(model); ?